streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -725,6 +725,37 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
         }
     }
 
+  def split(p : A => UIO[Boolean]) : ZManaged[R, E, (ZStream[R,E,A],ZStream[R,E,A])] = 
+    split(2)(p)
+
+  def split[B1,B2](queueCapacity : Int)(p : A => UIO[Boolean]) : ZManaged[R, E, (ZStream[R,E,A],ZStream[R,E,A])] = {
+    for {
+      leftQueue   <- ZManaged.make(Queue.bounded[Take[E, A]](queueCapacity))(_.shutdown)
+      rightQueue  <- ZManaged.make(Queue.bounded[Take[E, A]](queueCapacity))(_.shutdown)
+      folder      = (_:Unit, item:A) => {
+        p(item).flatMap {
+          case true   => leftQueue.offer(Take.Value(item)).fork.unit
+          case false  => rightQueue.offer(Take.Value(item)).fork.unit
+        }
+      } 
+      _           <- self.fold[R, E, A, Unit].flatMap { fold =>
```

Comment:
an equivalent (and simpler) way to do this is with `foreachManaged`:
```
self.foreachManaged { item => 
  // folder body here
}.foldCauseM(...).fork
```

Reply:
Do you mean replace `self.fold` with that i.e. starting at line **741**?

Reply:
yup